sdk/keyvault/keyvault-certificates/src/lro/create/operation.ts

```diff
@@ -0,0 +1,152 @@
+   */
+  initialResponse?: CertificateOperation;
+  /**
+   * @member {CertificateOperation} [previousResponse] The previous response received the last time the service was reached by the operation's update function
+   */
+  previousResponse?: CertificateOperation;
+  /**
+   * @member {Certificate} [pendingCertificate] The pending certificate, for easy access.
+   */
+  pendingCertificate?: Certificate;
+  /**
+   * @member {boolean} [doNotCreate] Boolean to skip the creation of the certificate.
+   */
+  doNotCreate?: boolean;
+}
+
+/**
+ * @interface
+ * An interface representing a create certificate's poll operation
+ */
+export interface CreateCertificatePollOperation
+  extends PollOperation<CreateCertificatePollOperationState, Certificate> {}
+
+/**
+ * @summary Reaches to the service and updates the create certificate's poll operation.
+ * @param [options] The optional parameters, which are an abortSignal from @azure/abort-controller and a function that triggers the poller's onProgress function.
+ */
+async function update(
+  this: CreateCertificatePollOperation,
+  options: {
+    abortSignal?: AbortSignalLike;
+    fireProgress?: (state: CreateCertificatePollOperationState) => void;
+  } = {}
+): Promise<CreateCertificatePollOperation> {
+  const state = this.state;
+  const { name, createCertificateOptions, client } = state;
+
+  const requestOptions = state.createCertificateOptions.requestOptions || {};
+  if (options.abortSignal) {
+    requestOptions.abortSignal = options.abortSignal;
+  }
+
+  const doFinalResponse = state.previousResponse && state.previousResponse.status !== "inProgress";
+
+  if (!state.initialResponse) {
+    if (!state.doNotCreate) {
+      let certificatePolicy = state.certificatePolicy || (await client.getCertificatePolicy(name));
+      await client.createCertificate(name, certificatePolicy!, createCertificateOptions);
+    }
+    state.pendingCertificate = await client.getCertificateWithPolicy(name, requestOptions);
+    state.previousResponse = await client.getPlainCertificateOperation(name, requestOptions);
+    state.initialResponse = state.previousResponse;
+    state.started = true;
+  } else if (doFinalResponse) {
+    state.previousResponse = await client.getPlainCertificateOperation(name, requestOptions);
+    state.result = await client.getCertificateWithPolicy(name, requestOptions);
+    state.completed = true;
+    state.pendingCertificate = undefined;
+  } else {
+    state.pendingCertificate = await client.getCertificateWithPolicy(name, requestOptions);
+    state.previousResponse = await client.getPlainCertificateOperation(name, requestOptions);
+  }
+
+  // Progress only after the poller has started and before the poller is done
+  if (state.initialResponse && !doFinalResponse && options.fireProgress) {
+    options.fireProgress(state);
```

Comment:
should we only fire this if the previousResponse differs from the response we just got in some meaningful way?